server/routes/urlshortener.js

```diff
@@ -1,20 +1,64 @@
 const { WINDOW_LOCATION_ORIGIN, PORT } = require('../config/constants');
-
 const { urlShorten } = require('../models/urlShorten');
 const { ShortenedUrlsRepository } = require('../services/shortenedUrlRepository');
 const { isUri } = require('../helper/web-url-validation');
-const { generate } = require('../middleware/shortid');
 const { IndexNotFoundError, NotFoundError } = require('../models/customErrors');
+const code = require('../helper/code');
 
 var urlRepository = new ShortenedUrlsRepository();
+
+var keys = [];
+var getSmallestIndex = () => {
+    // TODO: make smarter
+    if (keys.length == 0)  return 0;
+    keys.sort((a, b) => a - b);
+    return keys[keys.length - 1] + 1;
+}
+
 module.exports = app => {
+    app.get('/', function (req, res) {
+        return res.send(urlRepository.getAll());
+    })
+
+    // :url (in body) URL to shorten
+    app.post('/', function (req, res) {
+        const url = req.body['url'];
+
+
+        try {
+            isUri(url);
+        } catch (error) {
+            return res.status(400).send("error");
+        }
+
+        // if an url:
+        let index = getSmallestIndex();
+        keys.push(index); // TODO: This might be removed
+
+        var shortenedUrlId = code.encode(index);
+        const shortBaseUrl = WINDOW_LOCATION_ORIGIN + ':' + PORT;
+        let shortenedUrlObject = new urlShorten(url, shortBaseUrl + '/' + shortenedUrlId, shortenedUrlId);
+        shortenedUrlObject = urlRepository.add(shortenedUrlObject);
+
+        return res.status(201).json(shortenedUrlId);
+    })
+
+    app.delete('/', function (req, res) {
+        urlMap = {};
```

Comment:
What's this?

Reply:
whoops something I should delete :D my bad